### PR TITLE
RTEMIS-199: Added access check and workflow state change logic.

### PR DESCRIPTION
**Pull request by RohanSingh-inno**
Ticket # : https://innoraft.atlassian.net/browse/RTEMIS-199

---
### What the PR does
`(Explain fixes, changes and impacts)`
- Added access check for student performance.
- Added logic for workflow change for allocation entity.
- Remove required from some fields.

### CODE REVIEW

File: modules/rte_mis_student_tracking/config/install/field.field.mini_node.student_performance.field_final_rank.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ entity_type: mini_node
 bundle: student_performance
 label: 'Final Rank'
 description: 'Please add the final rank.'
-required: true
+required: false
```
Comment (kunal-sin) [inline modules/rte_mis_student_tracking/config/install/field.field.mini_node.student_performance.field_final_rank.yml:13]: @manish-3108 Just an FYI

File: modules/rte_mis_student_tracking/rte_mis_student_tracking.module [hunk 2]
```diff
@@ -213,3 +214,125 @@ function rte_mis_student_tracking_get_total_students() {
...
+      $details['gender'] = $student_details_mini_node->get('field_gender')->getString() ?? NULL;
+      $details['mobile_number'] = $student_details_mini_node->get('field_mobile_number')->value ?? NULL;
+      $details['religion'] = $student_details_mini_node->get('field_religion')->getString() ?? NULL;
+      $details['residential_address'] = $student_details_mini_node->get('field_residential_address')->getString() ?? NULL;
+
+      $student_gaurdian_name = $student_details_mini_node->get('field_guardian_name')->getString() ?? NULL;
+      $student_father_name = $student_details_mini_node->get('field_father_name')->getString() ?? NULL;
+      $student_mother_name = $student_details_mini_node->get('field_mother_name')->getString() ?? NULL;
+
+      $details['parent_name'] = $student_gaurdian_name ?? $student_father_name ?? $student_mother_name ?? NULL;
```
Comment (kunal-sin) [inline modules/rte_mis_student_tracking/rte_mis_student_tracking.module:252]: @RohanSingh-inno Can't we do this inside a loop?

File: modules/rte_mis_student_tracking/src/Access/PerformanceDetailEditAccessCheck.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        $performance_entity = $this->entityTypeManager->getStorage('mini_node')->load($mini_node->id());
+        $school = $performance_entity->get('field_school')->getString();
+        $academic_year = $performance_entity->get('field_academic_session')->getString();
+        $current_academic_year = _rte_mis_core_get_current_academic_year();
+        if (($school_id != NULL && $school_id != $school) || ($academic_year != $current_academic_year)) {
+          // Get the school details from user.
+          return AccessResult::forbidden()->setCacheMaxAge(0);
+        }
+        // Set cache max age to 0 for operation link in view to change.
+        return AccessResult::allowed()->setCacheMaxAge(0);
```
Comment (kunal-sin) [inline modules/rte_mis_student_tracking/src/Access/PerformanceDetailEditAccessCheck.php:54]: I don't think this statement is of any use. if the condition fails then line 57 will execute anyways.

File: modules/rte_mis_student_tracking/rte_mis_student_tracking.module (hunk 2 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_student_tracking/rte_mis_student_tracking.module]: yes we can do it.

File: modules/rte_mis_student_tracking/src/Access/PerformanceDetailEditAccessCheck.php (hunk 3 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_student_tracking/src/Access/PerformanceDetailEditAccessCheck.php]: sure.

**Pull request merged**